hail/src/main/scala/is/hail/rvd/RVDPartitioner.scala

```diff
@@ -14,7 +14,7 @@ class RVDPartitioner(
   // rangeBounds: Array[Interval[kType]]
   // rangeBounds is interval containing all keys within a partition
   val rangeBounds: Array[Interval],
-  allowedOverlap: Int
+  val allowedOverlap: Int
```

Comment:
This really shouldn't be considered part of the data of a partitioner. It's just a constructor argument to request a stronger precondition enforced. The reason it shouldn't be part of the data is that a given `rangeBounds` satisfies many different `allowedOverlap`, because smaller values imply all larger values, with `allowedOverlap = kType.size` always satisfied. The `allowedOverlap` that was enforced during construction is irrelevant to the behavior of the partitioner.